providers/vic/operations/pod_deleter_test.go

```diff
@@ -0,0 +1,233 @@
+	ip.On("Handle", op, podID, podName).Return(podHandle, nil)
+	ip.On("UnbindScope", op, podHandle, podName).Return(podHandle, fakeEP, nil)
+	ip.On("CommitHandle", op, podHandle, podID, int32(-1)).Return(nil)
+	ip.On("Remove", op, podID, true).Return(nil)
+
+	// Add vicPod to the cache
+	cache.Add(op, "", pod.Name, &vicPod)
+
+	// Failed SetState
+	fakeErr := fakeError("failed SetState")
+	ip.On("SetState", op, podHandle, podName, "STOPPED").Return("", fakeErr)
+	err = d.DeletePod(op, &pod)
+	assert.Equal(t, err, fakeErr, "Expected failed SetState error")
+}
+
+func TestDeletePodErrorCommitHandle(t *testing.T) {
+	client := client.Default
+	_, ip, cache, op := createMocks(t)
+
+	persona := "1.2.3.4"
+	portlayer := "1.2.3.4"
+
+	// Start with arguments
+	d, err := NewPodDeleter(client, ip, cache, persona, portlayer)
+	assert.NotNil(t, d, "Expected non-nil creating a pod Deleter but received nil")
+	assert.Nil(t, err, "Expected nil")
+
+	// Set up the mocks for this test
+	ip.On("Handle", op, podID, podName).Return(podHandle, nil)
+	ip.On("UnbindScope", op, podHandle, podName).Return(podHandle, fakeEP, nil)
+	ip.On("SetState", op, podHandle, podName, "STOPPED").Return(podHandle, nil)
+	ip.On("Remove", op, podID, true).Return(nil)
+
+	// Add vicPod to the cache
+	cache.Add(op, "", pod.Name, &vicPod)
+	// Failed Commit
+	fakeErr := fakeError("failed Commit")
+	ip.On("CommitHandle", op, podHandle, podID, int32(-1)).Return(fakeErr)
+	err = d.DeletePod(op, &pod)
+	assert.Equal(t, err, fakeErr, "Expected failed Commit error")
+}
+
+func TestDeletePodErrorRemove(t *testing.T) {
+	client := client.Default
+	_, ip, cache, op := createMocks(t)
+
+	persona := "1.2.3.4"
+	portlayer := "1.2.3.4"
+
+	// Start with arguments
+	d, err := NewPodDeleter(client, ip, cache, persona, portlayer)
+	assert.NotNil(t, d, "Expected non-nil creating a pod Deleter but received nil")
+	assert.Nil(t, err, "Expected nil")
+
+	// Set up the mocks for this test
+	ip.On("Handle", op, podID, podName).Return(podHandle, nil)
+	ip.On("UnbindScope", op, podHandle, podName).Return(podHandle, fakeEP, nil)
+	ip.On("SetState", op, podHandle, podName, "STOPPED").Return(podHandle, nil)
+	ip.On("CommitHandle", op, podHandle, podID, int32(-1)).Return(nil)
+
+	// Add vicPod to the cache
+	cache.Add(op, "", pod.Name, &vicPod)
+
+	// Failed Remove
+	fakeErr := fakeError("failed Remove")
+	ip.On("Remove", op, podID, true).Return(fakeErr)
+	err = d.DeletePod(op, &pod)
+	assert.Equal(t, err, fakeErr, "Expected failed Commit error")
+}
+
+func TestDeletePodErrorBadArgs(t *testing.T) {
+	client := client.Default
+	_, ip, cache, op := createMocks(t)
+	persona := "1.2.3.4"
+	portlayer := "1.2.3.4"
+
+	// Start with arguments
+	d, err := NewPodDeleter(client, ip, cache, persona, portlayer)
+	assert.NotNil(t, d, "Expected non-nil creating a pod Deleter but received nil")
+
+	// Negative Cases
+	err = d.DeletePod(op, nil)
+	assert.Equal(t, err, PodDeleterInvalidPodSpecError)
+}
+
+func init() {
```

Comment:
You should move this to the top of the file.  The reason I had it at the bottom of the file in the creator was because the function was so long.  Since you moved that code to the common file, we can follow the golang standard and put the init() up at the stop.